service/history/historyEngine.go

```diff
@@ -864,37 +883,49 @@ func (e *historyEngineImpl) queryDirectlyThroughMatching(
 
 func (e *historyEngineImpl) getMutableState(
 	ctx context.Context,
-	namespaceID namespace.ID,
-	execution commonpb.WorkflowExecution,
+	workflowKey definition.WorkflowKey,
 ) (_ *historyservice.GetMutableStateResponse, retError error) {
 
-	context, release, err := e.historyCache.GetOrCreateWorkflowExecution(
+	if len(workflowKey.RunID) == 0 {
```

Comment:
workflowConsistencyChecker.GetWorkflowContext can handle empty run id. Any reason to have this check?

Reply:
Caller of this function should already figure out the the current run ID, or the current run ID will be loaded from DB multiple times, e.g. poll current mutable state.